ui/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -823,6 +829,22 @@ export default class ConfirmTransactionBase extends Component {
       txData.contractMethodName = name;
     }
 
+    if (dappProposedTokenAmount) {
+      txData.dappProposedTokenAmount = dappProposedTokenAmount;
+      txData.originalApprovalAmount = dappProposedTokenAmount;
+    }
+
+    if (customTokenAmount) {
+      txData.customTokenAmount = customTokenAmount;
+      txData.finalApprovalAmount = customTokenAmount;
+    } else {
```

Comment:
maybe this should be `else if (dappProposedTokenAmount !== undefined)`, because we don't need to write undefined data to txData for `finalApprovalAmount`

Reply:
@danjm Yes, you are right, I will do that in the next commit.

Reply:
@danjm Done [here](https://github.com/MetaMask/metamask-extension/pull/15289/commits/00676999f69009ad1d132e9da0af0664cfaecc70).